### DISCUSSION TITLE
.NET Plugin capabilities

### DISCUSSION THREAD

**Discussion by Simply007** [dotansimha/graphql-code-generator#6518] [Q&A]
Hello,

I am DevRel Leader with [Kentico Kontent](https://kontent.ai) and we are developing [GraphQL support for our headless CMS Kentico Kontent](https://portal.productboard.com/kontent/2-kontent-public-roadmap/c/119-allow-developers-to-query-data-more-efficiently-using-graphql). I have a couple of questions about the C# plugin.

The first version of the generated types: https://github.com/Simply007/kontent-starter-graphql-net/blob/main/GraphQLWebStarter/Models/Generated/KontentTypes.cs

1. Is there a way to configure casing for property names? I would like to use PascalCase for property names so that it complies C# best practices + I have been testing the compatibility with `GraphQL.Client` C# client - https://github.com/Simply007/kontent-starter-graphql-net
2. Is there a way to programmatically hook to the generation process? I would need to use DateTime type for date&time element types instead of dynamic
3. Would it be possible to generate classes as `partial` to make it possible to extend them in a separate file?
4. What is the method `GetInputObject` for? Is there a way to prevent generating it multiple times?

I have tried to walk through the docs, but I haven't found the answers to these questions.

EDIT:
Linking an issue I have found as blocking showcasing the generator - GaphQL union in .NET -> should be fixed in #6858 (with PR #6862)

**Comment by ardatan** [comment]
1. We have `namingConvention` option to configure naming conventions to use pascal case, camel case or kebab case etc
2. You can hook into generation process using Lifecycle Hooks or you can use `scalars` option to configure what type should be used for a specific scalar type.
I am not familiar with C# plugin that much but those are my answers for the first two questions.
Also those configuration options can be found in here;
https://www.graphql-code-generator.com/docs/plugins/c-sharp
https://www.graphql-code-generator.com/docs/getting-started/lifecycle-hooks

**Reply by Simply007** [reply]
Hay @ardatan - thank you very much for the response. I have book Fridays for the research - that's why the delay.

1. I have been trying to play with the `namingConvention`, but it applies only to regions and classes names, but the properties are still in pascal case instead of pascal case.
2. The scalars worked - thx for pointing it out - I have needed to map DateTime + Guid - https://github.com/Simply007/kontent-starter-graphql-net/commit/499dac5fc67fb9f56873de9bf09a6a7cf9d90433

**Comment by Urigo** [comment]
just tagging some .NET past contributors..
@kurmeroli @cklam2 @aokht sorry it it's irrelevant
